dev.opengoal.OpenGOAL.yml

```diff
@@ -0,0 +1,88 @@
+id: dev.opengoal.OpenGOAL
+
+runtime: org.gnome.Platform
+runtime-version: '46'
+sdk: org.gnome.Sdk
+
+command: open-goal-launcher
+
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node20
+  - org.freedesktop.Sdk.Extension.rust-stable
+
+finish-args:
+  - --socket=wayland
+  - --socket=x11
```

Comment:
Why is both x11 and wayland uses? If the application does not support wayland just use x11, otherwise use fallback-x11 instead of x11.

Reply:
Ah I missed the description:

> I have enabled both Wayland and X11 as this is the only combination that works without graphical glitches in the launcher and can launch the games. Using only X11 left trailing black shadows graphical glitch when moving the launcher window, but games still start correctly. With wayland and x11-fallback, only game sound is played, controller works but no game window is created. I can set x11 as the only one if required, but it was not the best experience of all options in my opinion.

Reply:
> Using only X11 left trailing black shadows graphical glitch when moving the launcher window, but games still start correctly. 

Was this being used on a Wayland desktop environment session? Ie, the window was using Xwayland? Or in Xorg.

Sounds like this doesn't quite work with Xwayland/Wayland

Reply:
From my own testing this was what I thought was the best experience, as using only x11 the launcher worked, the game launch worked correctly, but anytime you moved the launcher window it left a trailing black shadow of the window until you stopped moving it. Using wayland and x11-fallback fixes the glitch but launching games did not work as you would only hear game sound no window was created for the game.

Having both wayland and x11 (no fallback) my system had no issues.

Reply:
> > Using only X11 left trailing black shadows graphical glitch when moving the launcher window, but games still start correctly.
> 
> Was this being used on a Wayland desktop environment session? Ie, the window was using Xwayland? Or in Xorg session?
> 
> Sounds like this doesn't quite work with Xwayland/Wayland

Yes, the current game renderer is not quite up to par yet. Work is being done on this currently

Reply:
> both wayland and x11

The wayland permission is entirely unused unless you/the toolkit is forcing/auto switching the window to be native wayland eg. GDK_BACKEND/MOZ_ENABLE_WAYLAND etc.

X11 socket gives you xwayland on a Wayland session and x11 window on xorg.

So this sounds weird as to why it'd cause any difference at all. I can test once the build is posted.

Reply:
> as using only x11 the launcher worked, the game launch worked correctly, but anytime you moved the launcher window it left a trailing black shadow of the window until you stopped moving it.

I can't reproduce this. Seems to be working fine on GNOME 46/Fedora 40 wayland session using only `x11`


https://github.com/flathub/flathub/assets/62639087/6a51e8f4-439d-43b2-b0d8-817ed3818cfa

Reply:
It is present on your recording just not as obvious on darker backgrounds.

Apologies for phone recording but I could not get it to show up on a screen recording. And like I said, it is minor but it is completely gone with wayland and x11 enabled. Other than that launcher and games work completely fine using only x11, just not wayland and x11 fallback.
[IMG_1535.webm](https://github.com/flathub/flathub/assets/26403463/a336a52f-aa4f-4b38-99e1-300acb598496)

Reply:
I think that was my VM being weird. Here it is on a lighter background in the VM.


https://github.com/flathub/flathub/assets/62639087/9de04f1f-6eb5-4a5a-8abc-33a8b5b3476b

Reply:
Ok well sounds like an issue on my machine. I'll remove it from the manifest